src/Microsoft.Azure.ServiceBus/Amqp/AmqpMessageConverter.cs

```diff
@@ -172,6 +173,14 @@ public static SBMessage AmqpMessageToSBMessage(AmqpMessage amqpMessage)
                 var byteArrayValue = (byte[])amqpMessage.ValueBody.Value;
                 sbMessage = new SBMessage(byteArrayValue);
             }
+            else if (amqpMessage.BodyType == SectionFlag.Data
```

Comment:
SectionFlag is over type [Flags]. So you will have to check that bit and not check for equality of entire object.
i.e., instead of `amqpMessage.BodyType == SectionFlag.Data`, you will have to do `(amqpMessage.BodyType & SectionFlag.AmqpValue) != 0`